### PR TITLE
External signal feature tests

### PR DESCRIPTION
**Pull request by mmcshane**
## What was changed

Add feature tests for go + py + ts + java for externally originated signals

## Why?

It's a system feature

## Checklist

1. Closes 

2. How was this tested:

n/a

3. Any docs updates needed?

No

### CODE REVIEW

File: features/signal/external/feature.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		)
+		return run, nil
+	},
+}
+
+// Workflow waits for a single signal and returns the data contained therein
+func Workflow(ctx workflow.Context) (string, error) {
+	wfResult := ""
+	signalCh := workflow.GetSignalChannel(ctx, SignalName)
+	selector := workflow.NewSelector(ctx)
```
Comment (cretz) [inline features/signal/external/feature.go:42]: You don't need a selector with only one arm, you can just receive on the channel directly

File: features/signal/external/feature.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+)
+
+const (
+	SignalName = "external_signal_channel"
+	SignalData = "Signaled!"
+)
+
+var Feature = harness.Feature{
+	Workflows:       Workflow,
+	WorkerOptions:   worker.Options{MaxHeartbeatThrottleInterval: 1 * time.Nanosecond},
```
Comment (Sushisource) [inline features/signal/external/feature.go:20]: Looks like you don't need to pass this option?

File: features/signal/external/feature.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import io.temporal.workflow.SignalMethod;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+@WorkflowInterface
+@ActivityInterface
```
Comment (Sushisource) [inline features/signal/external/feature.java:16]: No activities in here. There's a fix in my PR that allows init to work without this annotation / no activities.

**Review by Sushisource — APPROVED** [review]

File: features/signal/external/feature.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Override
+    public void externalSignal(String result) {
+      workflowResult = result;
+    }
+
+    @Override
+    public Run execute(Runner runner) throws Exception {
+      final var run = runner.executeSingleParameterlessWorkflow();
```
Comment (cretz) [inline features/signal/external/feature.java:41]: Nit: probably don't need `final`

**Review by cretz — APPROVED** [review]

**Comment by cretz** [comment]
Don't forget to run `npm run format` and, after you install [poe](https://github.com/nat-n/poethepoet), run `poe format`

File: features/signal/external/feature.java (hunk 4 above)
Comment (mmcshane) [reply features/signal/external/feature.java]: I'm a final zealot. But my kind is dying.

File: features/signal/external/feature.java (hunk 3 above)
Comment (mmcshane) [reply features/signal/external/feature.java]: Stole it. Whoever lands first wins?

Comment (cretz) [reply features/signal/external/feature.java]: I stole a Go fix from his review too and landed it. Sorry @Sushisource if I cause a conflict!

File: features/signal/external/feature.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  return result;
+}
+
+export const feature =
+  !wf.inWorkflowContext() &&
+  new Feature({
+    workflow,
+    execute: async (runner) => {
+      const handle = await runner.executeSingleParameterlessWorkflow();
+      handle.signal(externalSignal, signalData);
```
Comment (bergundy) [inline features/signal/external/feature.ts:26]: I see I'm late but you should await on this

Comment (bergundy) [reply features/signal/external/feature.ts:26]: We need a lint rule to catch this in this repo.
I'll add in my feature when I get time to work on it.

Comment (mmcshane) [reply features/signal/external/feature.ts:26]: Good catch, thank you. https://github.com/temporalio/sdk-features/pull/102

**Pull request merged**